src/main/java/cpw/mods/forge/serverpacklocator/secure/ProfileKeyPairBasedSecurityManager.java

```diff
@@ -0,0 +1,363 @@
+package cpw.mods.forge.serverpacklocator.secure;
+
+import com.mojang.authlib.exceptions.AuthenticationException;
+import com.mojang.authlib.minecraft.UserApiService;
+import com.mojang.authlib.yggdrasil.ServicesKeyInfo;
+import com.mojang.authlib.yggdrasil.YggdrasilAuthenticationService;
+import com.mojang.authlib.yggdrasil.response.KeyPairResponse;
+import cpw.mods.modlauncher.ArgumentHandler;
+import cpw.mods.modlauncher.Launcher;
+import io.netty.handler.codec.http.FullHttpRequest;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.lang.reflect.Field;
+import java.net.Proxy;
+import java.net.URLConnection;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.time.Instant;
+import java.time.format.DateTimeParseException;
+import java.util.Base64;
+import java.util.Objects;
+import java.util.UUID;
+
+public final class ProfileKeyPairBasedSecurityManager implements IConnectionSecurityManager
+{
+    private static final Logger LOGGER = LogManager.getLogger();
+    private static final ProfileKeyPairBasedSecurityManager INSTANCE = new ProfileKeyPairBasedSecurityManager();
+    private static final UUID DEFAULT_NILL_UUID = new UUID(0L, 0L);
+
+    public static ProfileKeyPairBasedSecurityManager getInstance()
+    {
+        return INSTANCE;
+    }
+
+    private final SigningHandler signingHandler;
+    private final UUID sessionId;
```

Comment:
I would rename this to something like `playerId` (and everywhere else in this class), because it's just the uuid of the player